nexus/src/db/datastore.rs

```diff
@@ -3394,6 +3396,93 @@ impl DataStore {
                 ))
             })
     }
+
+    pub async fn allocate_static_v6_address(
+        &self,
+        allocation_config: &StaticV6AllocationConfig,
+    ) -> Result<Ipv6Addr, Error> {
+        use db::schema::static_v6_address::dsl;
+
+        let allocation_config = allocation_config.clone();
+
+        self.pool()
+            .transaction(move |conn| {
+                // First, search for released addresses
+                let result: Result<StaticV6Address, diesel::result::Error> =
+                    dsl::static_v6_address
+                        .filter(dsl::allocated.eq(false))
+                        .limit(1)
+                        .get_result(conn);
+
+                if let Ok(row) = result {
+                    diesel::update(dsl::static_v6_address)
+                        .filter(dsl::address.eq(&row.address))
+                        .set(dsl::allocated.eq(true))
+                        .execute(conn)?;
+
+                    return Ok(row.address.parse().unwrap());
+                }
+
+                // If no released addresses exist, allocate a new one
+                // XXX there's got to be a better way! store the iterator
+                // somewhere?
```

Comment:
This should probably be stateless - remember, multiple Nexii could be executing simultaneously.

Reply:
@smklein Can you explain your concern a bit more? I believe the serializability guarantees of CRDB mean that we won't ever have multiple Nexii with the same address, even if they do concurrent operations. But I could be misunderstanding your point.

Reply:
I agree with everything you said - I mostly meant that "saving the in-memory iterator in Nexus to a variable" is a little risky, because if we have multiple Nexii, there would be multiple pieces of "in-memory" state that could conflict. We are definitely relying on the DB to resolve conflict and act as the source-of-truth.

This was all prompted by the comment `store the iterator somewhere?`.

Reply:
this is gone now